### PR TITLE
Add mitigations to HttpsClientCert_GetCertInformation flakiness

### PR DESCRIPTION
**Pull request by jkotalik**
For https://github.com/aspnet/IISIntegration/issues/1422.
- Deploy w3wp after creating the client certificate.
- Use unique name for cert issuer
- Log cert store and current cert on test failure.

### CODE REVIEW

File: test/Common.FunctionalTests/ClientCertificateTests.cs [hunk 1]
```diff
@@ -54,13 +57,14 @@ private async Task ClientCertTest(TestVariant variant, bool sendClientCert)
...
             deploymentParameters.AddHttpsToServerConfig();
 
-            var deploymentResult = await DeployAsync(deploymentParameters);
             var handler = new HttpClientHandler
             {
                 ServerCertificateCustomValidationCallback = (a, b, c, d) => true,
                 ClientCertificateOptions = ClientCertificateOption.Manual,
             };
 
+            var deploymentResult = await DeployAsync(deploymentParameters);
```
Comment (pakrym) [inline test/Common.FunctionalTests/ClientCertificateTests.cs]: What did this change?

**Comment by pakrym** [comment]
> Deploy w3wp after creating the client certificate. 

How is this accomplished?

File: test/Common.FunctionalTests/ClientCertificateTests.cs (hunk 1 above)
Comment (jkotalik) [reply test/Common.FunctionalTests/ClientCertificateTests.cs]: Sorry, meant to put this even lower. This is to make it so w3wp is create after creating the client handler.

File: test/Common.FunctionalTests/ClientCertificateFixture.cs [hunk 2]
```diff
@@ -12,72 +12,54 @@ public class ClientCertificateFixture : IDisposable
     {
         private X509Certificate2 _certificate;
         private string _certName;
```
Comment (pakrym) [inline test/Common.FunctionalTests/ClientCertificateFixture.cs]: You don't need a field.

File: test/Common.FunctionalTests/ClientCertificateFixture.cs [hunk 3]
```diff
@@ -91,7 +73,18 @@ public void Dispose()
...
+
+                // Remove any extra certs that were left by previous tests.
+                for (var i = store.Certificates.Count - 1; i >= 0; i--)
+                {
+                    var cert = store.Certificates[i];
+                    if (cert.Issuer.StartsWith(_certIssuerPrefix))
+                    {
+                        store.Remove(cert);
+                    }
+                    _certificate = cert;
```
Comment (pakrym) [inline test/Common.FunctionalTests/ClientCertificateFixture.cs]: Why are you setting this here?

File: samples/NativeIISSample/Properties/launchSettings.json [hunk 4]
```diff
@@ -14,8 +14,8 @@
       "commandLineArgs": "$(IISExpressArguments)",
       "environmentVariables": {
         "IIS_SITE_PATH": "$(MSBuildThisFileDirectory)",
-        "ANCM_PATH": "$(AncmPath)",
-        "ANCMV2_PATH": "$(AncmV2Path)",
+        "ANCM_PATH": "$(AspNetCoreModuleV1ShimDll)",
```
Comment (pakrym) [inline samples/NativeIISSample/Properties/launchSettings.json]: Revert or do it for all launchSettings.json

File: test/Common.FunctionalTests/ClientCertificateTests.cs [hunk 5]
```diff
@@ -54,31 +57,51 @@ private async Task ClientCertTest(TestVariant variant, bool sendClientCert)
...
+                }
             }
-            else
+            catch (Exception ex)
             {
-                Assert.Equal("Disabled", responseText);
+                Logger.LogError($"Certificate is invalid. Issuer name: {cert.Issuer}");
+                using (var store = new X509Store(StoreName.Root, StoreLocation.LocalMachine))
+                {
+                    Logger.LogError($"List of current certificates in root store: {cert.Issuer}");
```
Comment (pakrym) [inline test/Common.FunctionalTests/ClientCertificateTests.cs]: cert.Issuer?

File: samples/NativeIISSample/Properties/launchSettings.json (hunk 4 above)
Comment (jkotalik) [reply samples/NativeIISSample/Properties/launchSettings.json]: This was a missed file with https://github.com/aspnet/IISIntegration/commit/b0deed05be7710d80e28a1ce95ff4e760939644b

**Review by pakrym — APPROVED** [review]

**Pull request merged**